### ISSUE TITLE
5.8.1 fails to build: Error: Function 'ga_deallocate' at (1) has no IMPLICIT type; did you mean 'ga_allocate'?

### ISSUE THREAD

**Issue by yurivict** [GlobalArrays/ga#247]
```

 2454 |       if (.not.ga_deallocate(g_b))
      |               1
Error: Function 'ga_deallocate' at (1) has no IMPLICIT type; did you mean 'ga_allocate'?
global/src/scalapack.F:3125:19:

 3125 |            if(.not.ga_deallocate(g_b))
      |                   1
Error: Function 'ga_deallocate' at (1) has no IMPLICIT type; did you mean 'ga_allocate'?
global/src/scalapack.F:3180:17:

 3180 |          if(.not.ga_deallocate(g_b))
      |                 1
Error: Function 'ga_deallocate' at (1) has no IMPLICIT type; did you mean 'ga_allocate'?
global/src/scalapack.F:3405:17:

 3405 |          if(.not.ga_deallocate(g_b))
      |                 1
Error: Function 'ga_deallocate' at (1) has no IMPLICIT type; did you mean 'ga_allocate'?
global/src/scalapack.F:3451:20:

 3451 |             if(.not.ga_deallocate(g_b))
      |                    1
Error: Function 'ga_deallocate' at (1) has no IMPLICIT type; did you mean 'ga_allocate'?
global/src/scalapack.F:984:37:

  734 |      &       call ga_error('sl2gacpl: mem alloc failed A1 ', -1)
      |                                                             2
......
  984 |      &                               -info)
      |                                     1
Warning: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(4)/INTEGER(8)).
global/src/scalapack.F:3533:14:

 3506 |      3        dumm2, liwork4, info)
      |              2
......
 3533 |      3        int_mb(adriwork), liwork4, info)
      |              1
Warning: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
global/src/scalapack.F:4235:23:

 4200 |      3        descB, dumm, lwork4, dumm1,lrwork4, dumm2,
      |                                                  2
......
 4235 |      4        lrwork4, int_mb(adriwork), liwork4, info)
      |                       1
Warning: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
global/src/scalapack.F:4503:14:

 4472 |      3        dumm2, liwork4, info)
      |              2
......
 4503 |      3        int_mb(adriwork), liwork4, info)
      |              1
Warning: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/INTEGER(4)).
```

gcc10-10.3.0
OS: FreeBSD 13

**Comment by jeffhammond** [comment]
See #240

**Comment by edoapra** [comment]
The fixed 5.8.1 tarball has been available since January 19th 2022 as
https://github.com/GlobalArrays/ga/releases/download/v5.8.1/ga-5.8.1.tar.gz
